### PR TITLE
:ci — workflow_dispatch rescue trigger for [skip ci] merges

### PR DESCRIPTION
**Pull request by mikelward**
## Summary

Adds a `workflow_dispatch:` trigger to `ci.yml` so the release-AAB / Firebase / Play-upload pipeline can be re-fired manually when the natural `push` event was suppressed.

## Why

The bot's snapshot-regen commit uses `[skip ci]` in its subject (deliberate, prevents PR-branch CI loops). When that bot commit ends up as the head of a rebase-merge to `main`, GitHub natively skips workflow runs for the push, and the AAB never builds — Firebase/Play-internal-track receive nothing for that SHA. This is exactly what just happened on PR #233 → `088ebe0`.

## What changed

1. **New trigger:** `workflow_dispatch:` — accessible from the Actions UI (CI → "Run workflow" → target `main`).
2. **Gates broadened** on the six steps that publish artifacts (Firebase distribute, decode upload keystore, bundle release AAB, upload release AAB, prepare Play release notes, upload AAB to Play): `github.event_name == 'push'` → `(push || workflow_dispatch)`. Still gated on `github.ref == 'refs/heads/main'`, so dispatching from a feature branch is a no-op for the publishing steps.
3. **Fallbacks** for `github.event.head_commit.message` (which is null on workflow_dispatch):
    - Firebase release notes: GHA expression `head_commit.message || format('Manual workflow_dispatch run on {0}', github.sha)`.
    - Play "what's new" file: shell-side `if [ -z "$subject" ]; then subject=$(git log -1 --pretty=%s "$COMMIT_SHA"); fi`.

## Alternative considered

Removing `[skip ci]` from the snapshot-regen commit subject. The CLAUDE.md note already says GITHUB_TOKEN-authored commits don't trigger workflow runs, so `[skip ci]` is belt-and-suspenders for the PR-branch loop case. But removing it would mean every snapshot-regen-on-PR also fires CI on the *PR branch's* push event (alongside the existing `pull_request` event run) — same double-run problem the workflow's existing `branches: [main]` filter is trying to avoid. So `[skip ci]` stays useful on the PR branch; the rescue trigger is the cleaner fix for the merge-to-main edge case.

## Privacy

No change to what leaves the device. CI-only.

## Test plan

- [ ] PR CI passes (no behavior change for push / pull_request events).
- [ ] After merge, manually trigger from Actions UI on `main` and verify:
    - [ ] AAB builds at the current versionCode.
    - [ ] Firebase distribute step runs (release notes show "Manual workflow_dispatch run on …" if dispatched independently of a push).
    - [ ] Play upload step runs and the build appears in the Play Console internal track.
- [ ] Verify dispatching from a non-main branch is a no-op for the publishing steps (gates still require `refs/heads/main`).

---
_Generated by [Claude Code](https://claude.ai/code/session_014RXZGgaCWUhXMfyqjB5kJP)_

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,6 +8,15 @@ on:
...
   pull_request:
+  # Manual rescue trigger so the release-AAB / Firebase / Play-upload steps
+  # can be re-fired when the natural `push` event was suppressed. Common
+  # case: a PR's bot-authored snapshot-regen commit (`ci: regenerate UI
+  # snapshots [skip ci]`) ends up as the head commit of the rebase-merge to
+  # `main`, and GitHub natively skips workflow runs for any push whose head
+  # commit subject contains `[skip ci]`. Without this trigger, the AAB for
+  # that SHA never builds and the Play internal track never gets it. Run
+  # from Actions tab → CI → "Run workflow", target `main`.
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (mikelward) [reply .github/workflows/ci.yml:19]: Good catch — the publishing-step gates are step-level, but the job-level `env:` populates `FIREBASE_*` / `PLAY_*` on the runner regardless of whether the publish steps run, so a non-main dispatch with a tampered `ci.yml` could read them. Fixed in `910590f` (amended into the original commit) by adding a job-level guard on `android-build`:

```yaml
if: github.event_name != 'workflow_dispatch' || github.ref == 'refs/heads/main'
```

Effect: non-main `workflow_dispatch` skips the entire `android-build` job, so the secret-bearing job env never gets populated. push events (already restricted to main by `on:`) and pull_request events are unaffected. The unit-tests job has no secrets at job level, so it's left unguarded — a write-permission user can still manually run unit tests against a feature branch, which is a legitimate use.

Didn't go with the "split into a separate workflow + GitHub `environment` with required approvals" path: it's heavier, and the guard above closes the actual exfil window. Happy to add the environment+approvals layer in a follow-up if you want a tighter belt — let me know.

---
_Generated by [Claude Code](https://claude.ai/code)_
